geostore/s3_utils.py

```diff
@@ -52,3 +53,43 @@ def s3_url_reader(staging_url: str) -> GeostoreS3Response:
     staging_s3_client = get_s3_client_for_role(s3_role_arn)
     geostore_s3_client = get_s3_client_for_role(get_param(ParameterName.S3_USERS_ROLE_ARN))
     return s3_url_reader
+
+
+def get_s3_etag(s3_bucket: str, s3_object_key: str, logger: Logger) -> str:
+    geostore_s3_client = get_s3_client_for_role(get_param(ParameterName.S3_USERS_ROLE_ARN))
+
+    try:
+        s3_response = geostore_s3_client.head_object(Bucket=s3_bucket, Key=s3_object_key)
+        return s3_response["ETag"]
+    except ClientError as error:
+        if error.response["Error"]["Code"] != "404":
+            logger.debug(
+                f"Unable to fetch eTag for “{s3_object_key}” in s3://{s3_bucket} due to “{error}”",
+                extra={GIT_COMMIT: get_param(ParameterName.GIT_COMMIT)},
+            )
+        # rather than raise, we return an empty string, indicating that the etag is different
+        # thus allowing the next step to continue rather than stalling the entire process
+        return ""
```

Comment:
Usually we'd return `None` for an unknown value; meaning the function return value type would be `Optional[str]`. Then you won't have to do length tests to check whether you got an actual ETag back, you'd just do `s3_etag is not None`. Does that make sense?